File: 914-x-of-a-kind-in-a-deck-of-cards/914-x-of-a-kind-in-a-deck-of-cards.cpp
class Solution {
public:
    int gcd(int a, int b){
        if(b == 0) return a;
        return gcd(b, a%b);
    }
    
    bool hasGroupsSizeX(vector<int>& deck) {
        
        unordered_map<int, int> m;
        for(auto x: deck){
            m[x]++;
        }
        
        int res = m[deck[0]];
        
        for(auto it : m){
            int val = it.second;
            if(val == 1) return false;
            
            res = gcd(max(res, val), min(res, val));
        }
        
        return res > 1? true : false;
    }
};

File: 914-x-of-a-kind-in-a-deck-of-cards/914-x-of-a-kind-in-a-deck-of-cards.py
class Solution:
    def gcd(self, a, b):
        if b == 0:
            return a
        
        return self.gcd(b, a%b)
    
    def hasGroupsSizeX(self, deck: List[int]) -> bool:
        
        m = {}
        
        for x in deck:
            m[x] = m.get(x, 0) + 1
        
        res = m[deck[0]]
        
        for x in m.values():
            if x == 1:
                return False
            
            res = self.gcd(max(res, x), min(res, x))
        
        return True if res > 1 else False